### PR TITLE
[e2e] Add test: OpenShfit story behind proxy

### PR DESCRIPTION
**Pull request by jsliacan**
**Depends on:** PR #3417 

We start proxy in Before hook of each scenario and remove it in the After hook of each scenario. This way all openshift stories should directly work behind proxy. 

## Testing

Run e2e tests (this is part of them). Or run only this feature with `@linux && @story_openshift && @proxy` tag.

### CODE REVIEW

File: test/e2e/testsuite/testsuite.go [hunk 1]
```diff
@@ -238,7 +238,6 @@ func InitializeScenario(s *godog.ScenarioContext) {
 					fmt.Println(err)
 					os.Exit(1)
 				}
-
```
Comment (praveenkumar) [inline test/e2e/testsuite/testsuite.go]: @jsliacan I don't think just removing this line is doing what you described in the commit log.

Comment (jsliacan) [reply test/e2e/testsuite/testsuite.go]: Oh great, I must've mingled the commits when rearranging after last changes. Thanks for noticing.

**Comment by cfergeau** [comment]
This is almost a copy&paste from story_openshift.feature. It would be useful to have an indication of this in the commit log/in the file itself (as a reminder both have to be kept in sync). And some explanation why this is duplicated would not hurt (I guess reuse is hard with what we use for testing?)

**Comment by adrianriobo** [comment]
Indeed @cfergeau I think we should avoid the duplication, already discussed with @jsliacan, so we probably switch a bit the approach:

* We keep the simple proxy tests, so without knowing nothing about options on how to run tests, the default flow will cover the proxy setup.

* We will add a parameter on how to run the suite, so basically it will tell if we want to run everything with proxy, which basically means there will be an underneath logic on adding the proxy setup to the cluster on each spin of the cluster (this will allow us to run an in deep test of a cluster with a proxy setup)

**Comment by jsliacan** [comment]
Also, I asked on cucumber/godog about Examples (Outline) for the `Background` directive. We'll see what they say, maybe there's a way to I'm not seeing that they'll suggest and we can keep the setup transparent to the user and not duplicate things. I don't think they'll actually implement the feature. 

https://github.com/cucumber/godog/issues/567

Additionally, if you have suggestions about how to re-word my issue over there, I'm all ears. Thanks!

**Comment by gbraad** [comment]
how does this relate to #3684?

**Comment by jsliacan** [comment]
I'm closing this for now as the changes in here aren't substantial to keep and we anyways agreed on a different approach than duplicating a feature file.

**Pull request closed without merging**